cmd/podman/create.go

```diff
@@ -567,15 +567,24 @@ func parseCreateOpts(c *cli.Context, runtime *libpod.Runtime, imageName string,
 		workDir = data.Config.WorkingDir
 	}
 
-	// COMMAND
-	if len(command) == 0 {
-		command = data.Config.Cmd
-	}
-
 	// ENTRYPOINT
-	entrypoint := c.String("entrypoint")
-	if entrypoint == "" {
-		entrypoint = strings.Join(data.Config.Entrypoint, " ")
+	// User input entrypoint takes priority over image entrypoint
+	entrypoint := c.StringSlice("entrypoint")
+	if len(entrypoint) == 0 {
+		entrypoint = data.Config.Entrypoint
+	}
+
+	// Build the command
+	// If we have an entry point, it goes first
+	if len(entrypoint) > 0 {
+		command = entrypoint
+	}
+	if len(inputCommand) > 0 {
+		// User command overrides data CMD
+		command = append(command, inputCommand...)
```

Comment:
Is this right? I thought user command overrode ENTRYPOINT as well?

Reply:
It's correct.
```
 2018-02-10 13:22:22 ⌚  localhost in ~/tmp/entrypoint
○ → cat Dockerfile 
FROM alpine
ENTRYPOINT ["ls"]

 2018-02-10 13:22:31 ⌚  localhost in ~/tmp/entrypoint
○ → docker run -it --rm entrycmd:latest top
ls: top: No such file or directory
```

Reply:
Huh, never knew that. Cool.